conbench/app/_plots.py

```diff
@@ -413,59 +440,79 @@ def gen_js_callback_click_on_glyph_show_run_details(repo_string):
     )
 
 
-def time_series_plot(history, benchmark, run, height=380, width=1100):
-    # log.info("Time series plot for:\n%s", json.dumps(history, indent=2))
+def time_series_plot(
+    samples: List[HistorySample], benchmark, run, height=380, width=1100
+):
+    # log.info("Time series plot for:\n%s", json.dumps(samples, indent=2))
 
-    unit = history[0]["unit"]
-    with_dist = [h for h in history if h["distribution_mean"]]
-    formatted, axis_unit = _should_format(history, unit)
+    unit = samples[0].unit
+    with_dist = [s for s in samples if s.zscorestats.rolling_mean]
+    formatted, axis_unit = _should_format([s.mean for s in samples], unit)
     dist_change_indexes = [
-        ix for ix, result in enumerate(history) if result["begins_distribution_change"]
+        ix
+        for ix, sample in enumerate(samples)
+        if sample.zscorestats.begins_distribution_change
     ]
 
-    # Note(JP): `history` is an ordered list of dicts, each dict has a `mean`
+    # Note(JP): `samples` is an ordered list of dicts, each dict has a `mean`
     # key which is extracted here by default.
-    source_mean_over_time = _source(history, unit, formatted=formatted)
+    source_mean_over_time = _source(samples, unit, formatted=formatted)
 
     source_min_over_time = bokeh.models.ColumnDataSource(
         data=dict(
             # Insert NaNs to break the line at distribution changes
             x=_insert_nans(
-                util.tznaive_iso8601_to_tzaware_dt([x["timestamp"] for x in history]),
+                util.tznaive_iso8601_to_tzaware_dt(
+                    [s.commit_timestamp.isoformat() for s in samples]
+                ),
                 dist_change_indexes,
             ),
             # TODO: best-case is not always min, e.g. when data has a unit like
-            # bandwidth.
-            y=_insert_nans([min(x["data"]) for x in history], dist_change_indexes),
+            # bandwidth. Note(JP): min() must not have None passed as arg, that
+            # is why I have changed `s.data` to only contain type float
+            # elements (including math.nan, instead of None).
```

Comment:
This change created quite a bit of follow-up work across various  places in the code. But I think it was worth it. Simplification.

Reply:
Re-reading this I don't even know how I changed this. Did I? :) Well.